vnext/JSI/Shared/ChakraCoreRuntime.h

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#ifdef CHAKRACORE
+
+#include "ChakraCoreRuntimeArgs.h"
+#include "ChakraRuntime.h"
+
+#include "ChakraCoreDebugger.h"
+
+#include <memory>
+#include <string>
+
+namespace Microsoft::JSI {
+
+class ChakraCoreRuntime : public ChakraRuntime {
+ public:
+  ChakraCoreRuntime(const std::shared_ptr<ChakraCoreRuntimeArgs> &args) noexcept;
+  ~ChakraCoreRuntime() noexcept;
+
+#pragma region Functions_inherited_from_Runtime
+
+  facebook::jsi::Value evaluatePreparedJavaScript(
+      const std::shared_ptr<const facebook::jsi::PreparedJavaScript> &preparedJs) override;
+
+  std::string description() override;
+
+ private:
+  // We override the following because ChakraCore (unlike Chakra) offers
+  // weak reference semantics.
+  facebook::jsi::WeakObject createWeakObject(const facebook::jsi::Object &obj) override;
```

Comment:
>createWeakObject [](start = 28, length = 16)

Similar to comments on other code reviews, can we do "MakeWeakObject"? this would be consistent with Mso::Make and std::make_*

Reply:
This is a overriden method that we cannot change.

---
In reply to: [345519532](https://github.com/microsoft/react-native-windows/pull/3650#discussion_r345519532) [](ancestors = 345519532)